### ISSUE TITLE
top_n is broken, because min_rank is broken

### ISSUE THREAD

**Issue by perrystephenson** [thomasp85/tidygraph#70]
Both of these functions work correctly when the final column is the one you want to rank on. However they don't work with the `wt` argument.

I've tried to have a look at the **tidygraph** code and figure out where it went wrong, but no luck.

I've been able to work around it by doing this:

```
filter(rank(desc(n), ties.method = "min") <= 50)
```

**Comment by thomasp85** [comment]
Can I get you to provide a reproducible example illustrating this?

**Comment by perrystephenson** [comment]
Hey, sure thing.

Firstly, I create a dummy graph.
```
g <- play_erdos_renyi(10, 0.5) %>% 
  activate(nodes) %>% 
  mutate(degree = centrality_degree())
```

Then we can use `top_n(1)` without specifying a column, and it will pick the last column. All good.
```
g %>% top_n(1)
```
```
#> Selecting by degree
#> # A tbl_graph: 1 nodes and 0 edges
#> #
#> # A rooted tree
#> #
#> # Node Data: 1 x 1 (active)
#>   degree
#>    <dbl>
#> 1      8
#> #
#> # Edge Data: 0 x 2
#> # ... with 2 variables: from <int>, to <int>
```

Now I will make a new column on the end of the data frame, and do the same thing. Still working fine.
```
g %>% mutate(new_col = row_number()) %>% top_n(1)
```
```
#> Selecting by new_col
#> # A tbl_graph: 1 nodes and 0 edges
#> #
#> # A rooted tree
#> #
#> # Node Data: 1 x 2 (active)
#>   degree new_col
#>    <dbl>   <int>
#> 1      5      10
#> #
#> # Edge Data: 0 x 2
#> # ... with 2 variables: from <int>, to <int>
```

Now if I specify a column other than the final column, it recognises the argument (no message to the user) but doesn't do what it's told.
```
g %>% mutate(new_col = row_number()) %>% top_n(1, degree)
```
```
#> # A tbl_graph: 1 nodes and 0 edges
#> #
#> # A rooted tree
#> #
#> # Node Data: 1 x 2 (active)
#>   degree new_col
#>    <dbl>   <int>
#> 1      8       6
#> #
#> # Edge Data: 0 x 2
#> # ... with 2 variables: from <int>, to <int>
```

It's been a while since I looked at it, but IIRC I traced it back to a bug in `min_rank` before giving up and working around the issue.

**Comment by perrystephenson** [comment]
Ignore my previous (deleted) comment. It looks like this has been fixed, and I don't have access to the computer I was using when I raised this issue to double check the environment. It might have been fixed by something you did (although I can't see any commits that would have fixed this), a version mismatch with **dplyr** that has been resolved with a fresh install, or good old user error on my part.
